lib/src/chart/line_chart/line_chart_painter.dart

```diff
@@ -84,10 +83,19 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
       canvas.saveLayer(Rect.fromLTWH(0, -40, size.width + 40, size.height + 40), Paint());
     }
 
+    super.superBorderPaint(canvas, size);
+    super.drawBackground(canvas, size);
+    drawRangeAnnotation(canvas, size);
```

Comment:
Hi, I've checked it but I have a problem with it,
there is a big complexity here, I think we can simplify it.
these logics are applied only in the line chart,
and I think in the near future we need the range annotations for the bar charts,
then why not we handle it in the AxisChartPainter class, and don't touch the painting order?
we can have exact this paint order in the superclass, instead of exposing superclass paint functions,
the only drawback is moving `RangeAnnotations` class into the `AxisChartData` and handle the paint function in the `AxisChartPainter` class,
it makes breaking changes, but I think it is necessary.
What do you think?

Reply:
I'd say if you want to merge this and then enhance it to a more ideal state later I don't see why not but there's no reason not to merge it right now because this PR fixes bugs immediately

Reply:
Yes, it fixes the issue (and I think it's better to have an individual issue and mention it here to track it), 
but If there is a clearly better way to do that, we should merge it in the proper way,
we know the issue, we know the immediate solution, and we know the best solution.
then please let to do it in the best solution’s way.
If you are frustrated, I'm so sorry about that, but we need to keep our code clean and readable.
I can continue it if you let me.
Thanks!

Reply:
Sure I have no problem with that I was just saying that the branch could be merged twice,  one to get it in production now and one to enhance it to where you want it in the future. I will rebase these commits out of my other PR.

Reply:
No worries, 
I will do it tonight ASAP.
Thanks!